src/osd/PGLog.h

```diff
@@ -1258,6 +1257,7 @@ struct PGLog : DoutPrefixProvider {
     IndexedLog *log,
     missing_type &missing,
     LogEntryHandler *rollbacker,
+    std::optional<eversion_t> roll_forward_to,
```

Comment:
how about using a bool `should_rollforward` and setting it to true when `log->add(*p);` happens and check `should_rollforward` in `merge_log` before doing `log.roll_forward_to`?

Reply:
See above, that was my original thought, but I think if we just remove this param handling it entirely within this method should be fine.

Reply:
At least, it seems appropriate to me for the logic that rolls the log forward to immediately update crt.